libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/MapboxTripStarter.kt

```diff
@@ -0,0 +1,197 @@
+package com.mapbox.navigation.core.trip
```

Comment:
As you described in the class description. _"This makes it simpler to start a MapboxNavigation trip session.'_ 
It is a convenience over the existing MapboxNavigation API. Therefore , I think we should move this to a higher-level module like `libnavui-base`.

Reply:
Moving it there will also allow us to use `UIComponent` as the base class.

Reply:
> libnavui-base

Putting it in base means it is cannot be used by solutions that only have the core sdk. `com.mapbox.navigation:core`. Higher level classes are going in core too, for example the `ReplayRouteSession` is in core.

Difference between `core` and `ui` is not always low level vs high level, it is often "visual" vs "logical". The `MapboxTripStarter` logic may be better inside `MapboxNavigation` but am building it on the outside so that it carries less risk for regressions.